tools/chplcheck/src/chplcheck.py

```diff
@@ -263,8 +263,13 @@ def main():
 
     printed_warning = False
 
+    prev_context = None
     for filename, context in chapel.files_with_contexts(args.filenames):
-        context.set_module_paths([], [])
+        # Avoid re-setting module paths on already-used contexts, due to
+        # bucketing from files_with_contexts.
+        if context is not prev_context:
+            context.set_module_paths([], [])
+        prev_context = context
```

Comment:
Hmm, what about an (optional) argument to `files_with_contexts` that's a lambda to execute on a context object? It seems to make the most sense to provide that as a "way to configure the context on creation".

```Python
for filename, context in chapel.files_with_contexts(args.filenames, lambda context: context.set_module_paths([], [])):
```

Reply:
I'm thinking something like this: https://github.com/chapel-lang/chapel/pull/25326

Reply:
I like that, will adjust to use `files_with_stdlib_contexts`.